### PR TITLE
CameraGimbalSliderBar

### PR DESCRIPTION
**Pull request by leesuann**
CameraGimbalSliderBar

image

### CODE REVIEW

File: custom/qgroundcontrol.qrc [hunk 1]
```diff
@@ -290,8 +290,17 @@
...
         <file alias="QGroundControl/Controls/CameraBottomMenu.qml">../src/QmlControls/CameraBottomMenu.qml</file>
+		<file alias="QGroundControl/Controls/SdCardDialog.qml">../src/QmlControls/SdCardDialog.qml</file>
+		<file alias="QGroundControl/Controls/CameraGimbalSliderBar.qml">../src/QmlControls/CameraGimbalSliderBar.qml</file>
+		<file alias="QGroundControl/Controls/AntiFlicker.qml">../src/QmlControls/AntiFlicker.qml</file>
+		<file alias="QGroundControl/Controls/LearningCenterDialog.qml">../src/QmlControls/LearningCenterDialog.qml</file>
+		<file alias="QGroundControl/Controls/PreflightSecurityCheck.qml">../src/QmlControls/PrelightSecurityCheck.qml</file>
+		<file alias="QGroundControl/Controls/PreflightSecurityCheckV1.qml">../src/QmlControls/PrelightSecurityCheckV1.qml</file>
+		<file alias="QGroundControl/Controls/RemoteControlInstruction.qml">../src/QmlControls/RemoteControlInstruction.qml</file>
+		<file alias="QGroundControl/Controls/VideoEncodingFormat.qml">../src/QmlControls/VideoEncodingFormat.qml</file>
+		<file alias="QGroundControl/Controls/SaveLocation.qml">../src/QmlControls/SaveLocation.qml</file>
```
Comment (thanhkaist) [inline custom/qgroundcontrol.qrc:303]: why do you add these files? Also, is there any reason to modify the alias path to QGroundControl/Controls/  (e.g: original PreflightSecurityCheckV1.qml ->   QGroundControl/Controls/PreflightSecurityCheckV1.qml )

File: custom/qgroundcontrol.qrc [hunk 2]
```diff
@@ -290,8 +290,17 @@
         <file alias="QGroundControl/Controls/CameraRecordButton.qml">../src/QmlControls/CameraRecordButton.qml</file>
         <file alias="QGroundControl/Controls/CameraGalleryButton.qml">../src/QmlControls/CameraGalleryButton.qml</file>
         <file alias="QGroundControl/Controls/CommonTopMenu.qml">../src/QmlControls/CommonTopMenu.qml</file>
-        <file>MissionHistoryItem.qml</file>
+        <file alias="QGroundControl/Controls/MissionHistoryItem.qml">../src/QmlControls/MissionHistoryItem.qml</file>
```
Comment (thanhkaist) [inline custom/qgroundcontrol.qrc:293]: Why do you add this item in this  file?

File: qgroundcontrol.qrc [hunk 3]
```diff
@@ -296,18 +296,20 @@
         <file alias="CameraGalleryButton.qml">src/QmlControls/CameraGalleryButton.qml</file>
         <file alias="QGroundControl/Controls/Grid.qml">src/QmlControls/Grid.qml</file>
         <file alias="QGroundControl/Controls/CenterPoint.qml">src/QmlControls/CenterPoint.qml</file>
-        <file alias="CommonTopMenu.qml">src/QmlControls/CommonTopMenu.qml</file>
+        <file alias="QGroundControl/Controls/CommonTopMenu.qml">src/QmlControls/CommonTopMenu.qml</file>
```
Comment (thanhkaist) [inline qgroundcontrol.qrc:299]: What is the reason behind this changing path?

File: qgroundcontrol.qrc [hunk 4]
```diff
@@ -296,18 +296,20 @@
...
-        <file alias="PreflightSecurityCheck.qml">src/QmlControls/PrelightSecurityCheck.qml</file>
-        <file alias="PreflightSecurityCheckV1.qml">src/QmlControls/PrelightSecurityCheckV1.qml</file>
-        <file alias="RemoteControlInstruction.qml">src/QmlControls/RemoteControlInstruction.qml</file>
-        <file alias="SdCardDialog.qml">src/QmlControls/SdCardDialog.qml</file>
-        <file alias="MissionHistoryItem.qml">src/QmlControls/MissionHistoryItem.qml</file>
+		<file alias="QGroundControl/Controls/CameraBottomMenu.qml">src/QmlControls/CameraBottomMenu.qml</file>
+        <file alias="QGroundControl/Controls/LearningCenterDialog.qml">src/QmlControls/LearningCenterDialog.qml</file>
+        <file alias="QGroundControl/Controls/PreflightSecurityCheck.qml">src/QmlControls/PrelightSecurityCheck.qml</file>
+        <file alias="QGroundControl/Controls/PreflightSecurityCheckV1.qml">src/QmlControls/PrelightSecurityCheckV1.qml</file>
+        <file alias="QGroundControl/Controls/RemoteControlInstruction.qml">src/QmlControls/RemoteControlInstruction.qml</file>
```
Comment (thanhkaist) [inline qgroundcontrol.qrc:306]: What is the reason behind this changing path?

File: qgroundcontrol.qrc [hunk 5]
```diff
@@ -296,18 +296,20 @@
...
+        <file alias="QGroundControl/Controls/SdCardDialog.qml">src/QmlControls/SdCardDialog.qml</file>
+        <file alias="QGroundControl/Controls/MissionHistoryItem.qml">src/QmlControls/MissionHistoryItem.qml</file>
```
Comment (thanhkaist) [inline qgroundcontrol.qrc:308]: What is the reason behind this changing path?

File: src/QmlControls/CameraGimbalSliderBar.qml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+import QGroundControl.Controls              1.0
+import QGroundControl.Palette               1.0
+import QGroundControl.MultiVehicleManager   1.0
+import QGroundControl.ScreenTools           1.0
+import QGroundControl.Controllers           1.0
+import QtGraphicalEffects                   1.12
+
+Rectangle {
+    id: root
+    z : 3
```
Comment (thanhkaist) [inline src/QmlControls/CameraGimbalSliderBar.qml]: Avoid using z

File: src/QmlControls/CameraGimbalSliderBar.qml [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    property color transparent: "transparent"
+    property color fontColorlightGray: "lightGray"
+
+    implicitWidth: defaultFontSize * 6
+    implicitHeight:defaultFontSize * 60
+
+    color: transparent
+    Column {
+        spacing: 0
+        z: 5
```
Comment (thanhkaist) [inline src/QmlControls/CameraGimbalSliderBar.qml]: Avoid using z

File: src/QmlControls/CameraGimbalSliderBar.qml [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+            }
+
+            handle: Item {
+                width: control.availableWidth
+                height: defaultFontSize * 7
+                anchors.horizontalCenter: bgRect.horizontalCenter
+                y: control.visualPosition * (control.availableHeight - height + defaultFontSize)
+
+                Image {
+                    source: "/res/CameraGimbalControl.svg"
```
Comment (thanhkaist) [inline src/QmlControls/CameraGimbalSliderBar.qml]: Use better path for images (e.g "qrc:/res/CameraGimbalControl.svg"). Correct other path also

File: src/QmlControls/CameraGimbalSliderBar.qml [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+            }
+        }
+
+        Button {
+            width: control.availableWidth
+            height: width
+            background: Rectangle { color: transparent }
+
+            Image {
+                source: "/res/CameraGimbal90.svg"
```
Comment (thanhkaist) [inline src/QmlControls/CameraGimbalSliderBar.qml]: Use better path for images (e.g "qrc:/res/CameraGimbalControl.svg"). Correct other path also

File: qgroundcontrol.qrc (hunk 4 above)
Comment (leesuann) [reply qgroundcontrol.qrc:306]: The files in other QmlControls folders are named the following, so they are just set the same.

File: qgroundcontrol.qrc (hunk 5 above)
Comment (leesuann) [reply qgroundcontrol.qrc:308]: The files in other QmlControls folders are named the following, so they are just set the same.

File: custom/qgroundcontrol.qrc (hunk 2 above)
Comment (leesuann) [reply custom/qgroundcontrol.qrc:293]: image

This error kept coming up, so I changed the path to be the same as the other file.

File: qgroundcontrol.qrc (hunk 3 above)
Comment (leesuann) [reply qgroundcontrol.qrc:299]: I set it to the same naming convention as the other files.

File: custom/qgroundcontrol.qrc (hunk 1 above)
Comment (leesuann) [reply custom/qgroundcontrol.qrc:303]: The files in other QmlControls folders are named the following, so they are just set the same.

**Review by thanhkaist — APPROVED** [review]

**Pull request merged**